packages/intl/src/MessageFormat/Parser/Parser.php

```diff
@@ -0,0 +1,627 @@
+<?php
+
+namespace Tempest\Intl\MessageFormat\Parser;
+
+use Tempest\Intl\MessageFormat\Parser\Node\ComplexBody\ComplexBody;
+use Tempest\Intl\MessageFormat\Parser\Node\ComplexBody\Matcher;
+use Tempest\Intl\MessageFormat\Parser\Node\ComplexBody\SimplePatternBody;
+use Tempest\Intl\MessageFormat\Parser\Node\ComplexBody\Variant;
+use Tempest\Intl\MessageFormat\Parser\Node\ComplexMessage;
+use Tempest\Intl\MessageFormat\Parser\Node\Declaration\InputDeclaration;
+use Tempest\Intl\MessageFormat\Parser\Node\Declaration\LocalDeclaration;
+use Tempest\Intl\MessageFormat\Parser\Node\Expression\Attribute;
+use Tempest\Intl\MessageFormat\Parser\Node\Expression\Expression;
+use Tempest\Intl\MessageFormat\Parser\Node\Expression\FunctionCall;
+use Tempest\Intl\MessageFormat\Parser\Node\Expression\FunctionExpression;
+use Tempest\Intl\MessageFormat\Parser\Node\Expression\LiteralExpression;
+use Tempest\Intl\MessageFormat\Parser\Node\Expression\Option;
+use Tempest\Intl\MessageFormat\Parser\Node\Expression\VariableExpression;
+use Tempest\Intl\MessageFormat\Parser\Node\Identifier;
+use Tempest\Intl\MessageFormat\Parser\Node\Key\WildcardKey;
+use Tempest\Intl\MessageFormat\Parser\Node\Literal\Literal;
+use Tempest\Intl\MessageFormat\Parser\Node\Literal\QuotedLiteral;
+use Tempest\Intl\MessageFormat\Parser\Node\Literal\UnquotedLiteral;
+use Tempest\Intl\MessageFormat\Parser\Node\Markup\Markup;
+use Tempest\Intl\MessageFormat\Parser\Node\Markup\MarkupType;
+use Tempest\Intl\MessageFormat\Parser\Node\MessageNode;
+use Tempest\Intl\MessageFormat\Parser\Node\ParsingException;
+use Tempest\Intl\MessageFormat\Parser\Node\Pattern\Pattern;
+use Tempest\Intl\MessageFormat\Parser\Node\Pattern\Placeholder;
+use Tempest\Intl\MessageFormat\Parser\Node\Pattern\QuotedPattern;
+use Tempest\Intl\MessageFormat\Parser\Node\Pattern\Text;
+use Tempest\Intl\MessageFormat\Parser\Node\SimpleMessage;
+use Tempest\Intl\MessageFormat\Parser\Node\Variable;
+
+final class Parser
```

Comment:
Could we rename this to MessageFormatParser or something alike? I like class names that are identifiable regardless of the namespace, especially class names that are as generic as "parser"